src/inputs/input_provider.jl

```diff
@@ -0,0 +1,26 @@
+"""
+    $TYPEDEF
+
+Acts as an interface between a given `InputFields` container and one or more `AbstractInputSource`s
+that are responsible for intiializing and updating the input `Field`s at each time step.
+"""
+struct InputProvider{
+    NF,
+    Grid<:AbstractLandGrid{NF},
+    Inputs<:InputFields{NF, Grid},
+    SourceType<:AbstractInputSource
+}
+    "Input variable `Field` states"
+    fields::Inputs
+
+    "A list of input sources to be invoked by `update_inputs!`"
+    sources::Vector{SourceType}
+end
+
+InputProvider(grid::AbstractLandGrid, sources::AbstractInputSource...) = InputProvider(InputFields(grid), collect(sources))
+
+function update_inputs!(provider::InputProvider, clock::Clock)
+    for source in provider.sources
+        update_inputs!(provider.fields, source, clock)
```

Comment:
I was thinking back and forth a bit, I think I am good with a container format here to interface between the input sources and the variables, especially for situations in which we have more than one input source. We should probably have a convience constructor for this either here or in the Simulation though so that users don't have to do this in most simulation and can also only hand over a `InputSource`